src/lib.rs

```diff
@@ -42,6 +43,43 @@ pub use dataset::{
 pub use driver::Driver;
 pub use metadata::Metadata;
 
+/// Apply GeoTransform to x/y coordinate.
```

Comment:
The `GeoTransform` relevant methods might be better placed inside the `dataset` or the `rasters` module.  It is anyway only relevant to rasters IIUC.